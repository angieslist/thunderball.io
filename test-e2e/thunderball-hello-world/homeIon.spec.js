import fetch from 'node-fetch';

function normalizeText(text) {
  // normalize hashed assets
  return text.replace(/[\d\w]{7,}(?=\.js|\.css)/g, 'TEST-HASH')
    // normalize app version
    .replace(/appVersion":"\d+\.\d+\.\d+/, 'appVersion":"TEST-VERSION')
    // normalize data-react-checksum
    .replace(/-?\d{7,}/g, 'TEST-CHECKSUM');
}

function getHeadersObject(headers) {
  const headersObj = {};
  headers.forEach((value, key) => {
    headersObj[key] = value;
  });
  return headersObj;
}

describe('homeIon', () => {
  describe('Get / (HomeView)', () => {
    it('returns the page', async () => {
      const response = await fetch('http://localhost:8000');
      expect(response).toHaveProperty('status', 200);
      const headers = getHeadersObject(response.headers);
      expect(headers).toMatchObject({
        connection: 'close',
        'content-encoding': 'gzip',
        'content-type': 'text/html; charset=utf-8',
        date: expect.anything(),
        etag: expect.anything(),
        'transfer-encoding': 'chunked',
        vary: 'Accept-Encoding',
        'x-powered-by': 'Thunderball',
      });
      const text = normalizeText(await response.text());
      expect(text).toMatchSnapshot();
    });
  });
});
