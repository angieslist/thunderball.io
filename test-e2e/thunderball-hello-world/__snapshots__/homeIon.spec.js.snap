// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`homeIon Get / (HomeView) returns the page 1`] = `

<!DOCTYPE html>
<html lang="en">
  <head>
    <title data-rh="true">
      Home - Thunderball Hello World
    </title>
    <link data-rh="true"
          rel="shortcut icon"
          type="image/png"
          href="/assets/fb8bTEST-CHECKSUMe246c3363bfcc4f601.png"
    >
    <link data-rh="true"
          rel="stylesheet"
          href="//fonts.googleapis.com/css?family=Roboto:300,300italic,700,700italic"
    >
    <link data-rh="true"
          rel="stylesheet"
          href="//cdn.rawgit.com/necolas/TEST-HASH.css/master/TEST-HASH.css"
    >
    <link data-rh="true"
          rel="stylesheet"
          href="//cdn.rawgit.com/milligram/milligram/master/dist/milligram.min.css"
    >
    <link href="/assets/commons-TEST-HASH.css"
          rel="stylesheet"
          key="0"
    >
    <link href="/assets/home-TEST-HASH.css"
          rel="stylesheet"
          key="1"
    >
  </head>
  <body>
    <script type="text/javascript">
      window.__INITIAL_STATE__ = {"config":{"appVersion":"TEST-VERSION","appName":"thunderball-hello-world"},"device":{"host":"http:\\u002F\\u002Flocalhost:8000"},"routing":{"locationBeforeTransitions":{"pathname":"\\u002F","search":"","hash":"","action":"POP","key":null,"query":{}}}};
    </script>
    <div id="app">
      <div data-reactroot>
        <div class="container">
          <div class="row">
            <div class="column">
              <div>
                <div>
                  <h1>
                    Welcome to Thunderball Hello World
                  </h1>
                  <p>
                    This is an example of what you can do with Thunderball to help you get started.
                  </p>
                  <h2>
                    Other Thunderball Ions in this app
                  </h2>
                  <p>
                    You can add separate React apps as Thunderball ions
                  </p>
                  <ul>
                    <li>
                      <a href="/todos">
                        TODO App
                      </a>
                    </li>
                  </ul>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <script type="text/javascript"
            src="https://cdn.polyfill.io/v2/polyfill.min.js?features=Intl.~locale.en"
    >
    </script>
    <script type="text/javascript"
            src="/assets/commons-TEST-HASH.js"
    >
    </script>
    <script type="text/javascript"
            src="/assets/home-TEST-HASH.js"
    >
    </script>
  </body>
</html>

`;
